.github/actions/setup-distro/action.yaml

```diff
@@ -155,107 +131,71 @@ runs:
           exit 1
         }
 
-    - name: Configure ${{ inputs.distro-name }} to support cloud-init (hacks)
+    - name: Setup ${{ inputs.distro-name }} WSL instance
       shell: pwsh
-      if: ${{ fromJSON(inputs.wsl-enable-cloud-init-hack) }}
       env:
         WSL_DISTRO_NAME: ${{ inputs.wsl-distro-name }}
       run: |
         Set-StrictMode -version latest
 
-        Write-Output "Adding ubuntu-wsl-dev PPA to ${Env:WSL_DISTRO_NAME}..."
-        $wslArgs=@(
-          "--distribution ${Env:WSL_DISTRO_NAME}"
-          "--user root"
-          "--cd /"
-          "--exec add-apt-repository -y ppa:ubuntu-wsl-dev/ppa"
-        )
-        Start-Process -NoNewWindow -Wait -FilePath "wsl.exe" -ArgumentList $wslArgs
-        if ( ! $? ) {
-          Write-Error "Cannot add ubuntu-wsl-dev PPA to ${Env:WSL_DISTRO_NAME}."
-          exit 1
+        if ("$${{ inputs.wsl-enable-systemd }}" -eq "true" ) {
+          Write-Output "Terminating ${Env:WSL_DISTRO_NAME} for earlier changes to take effect..."
+          Start-Process -NoNewWindow -Wait -FilePath "wsl.exe" -ArgumentList "--terminate ${Env:WSL_DISTRO_NAME}"
```

Comment:
magic :magic_wand:

Reply:
That's a bit mouthful, yeah ;)

Here we basically shut down the instance so that it boots with systemd next.